src/server/routes/admin/nonces.ts

```diff
@@ -0,0 +1,169 @@
+      onchainNonce: Type.String({
+        description: "Last mined nonce",
+        examples: ["0"],
+      }),
+      lastUsedNonce: Type.String({
+        description: "Last incremented nonce sent to the blockchain",
+        examples: ["0"],
+      }),
+      sentNonces: Type.Array(
+        Type.String({
+          examples: ["0"],
+        }),
+        {
+          description:
+            "Nonces that were successfully sent to the blockchain but not mined yet.",
+        },
+      ),
+      recycledNonces: Type.Array(
+        Type.String({
+          examples: ["0"],
+        }),
+        {
+          description:
+            "Nonces that were acquired but failed to be sent to the blockchain, waiting to be recycled or cancelled.",
+        },
+      ),
+      chainId: Type.Number({
+        description: "Chain ID",
+        examples: [80001],
+      }),
+    }),
+  ),
+});
+
+responseBodySchema.example = {
+  result: [
+    {
+      walletAddress: "0xcedf3b4d8f7f1f7e0f7f0f7f0f7f0f7f0f7f0f7f",
+      onchainNonce: "2",
+      lastUsedNonce: "8",
+      recycledNonces: ["6", "7"],
+      chainId: 80001,
+    },
+  ],
+};
+
+export async function getNonceDetailsRoute(fastify: FastifyInstance) {
+  fastify.route<{
+    Querystring: Static<typeof walletWithAddressParamSchema>;
+    Reply: Static<typeof responseBodySchema>;
+  }>({
+    method: "GET",
+    url: "/admin/nonces",
+    schema: {
+      summary: "Get transaction details",
+      description: "Get raw logs and details for a transaction by queueId.",
+      tags: ["Admin"],
+      operationId: "transactionDetails",
+      querystring: walletWithAddressParamSchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseBodySchema,
+      },
+      hide: true,
+    },
+    handler: async (request, reply) => {
+      const { walletAddress, chain } = request.query;
+      const result = await getNonceDetails({ walletAddress, chainId: chain });
+
+      reply.status(StatusCodes.OK).send({
+        result,
+      });
+    },
+  });
+}
+
+export const getNonceDetails = async ({
+  walletAddress,
+  chainId,
+}: {
+  walletAddress?: string;
+  chainId?: string;
+} = {}) => {
+  const lastUsedNonceKeys = await getLastUsedNonceKeys(walletAddress, chainId);
+
+  const result = await Promise.all(
+    lastUsedNonceKeys.map(async (key) => {
+      const { chainId, walletAddress } = splitLastUsedNonceKey(key);
+
+      const [lastUsedNonce, sentNonces, recycledNonces, onchainNonce] =
+        await Promise.all([
+          redis.get(lastUsedNonceKey(chainId, walletAddress)),
+          redis.smembers(sentNoncesKey(chainId, walletAddress)),
+          redis.smembers(recycledNoncesKey(chainId, walletAddress)),
+          getOnchainNonce(chainId, walletAddress),
+        ]);
+
+      return {
+        walletAddress,
+        onchainNonce: onchainNonce.toString(),
+        lastUsedNonce: lastUsedNonce ?? "0",
+        sentNonces,
+        recycledNonces,
```

Comment:
Would be nice to sort these, hence using numbers

Reply:
makes sense. I would think offloading the compute to redis would be more efficient (?) so when we move to sorted sets we will receive them sorted anyway

Reply:
IMO don't worry about efficiency. string -> number is free. Worry about code cleanliness over performance.

Especially if we're going to sort in the dashboard. I'd rather our backend cast the type + sort than the client's browser.